src/routes/tools/_common/helpers.js

```diff
@@ -234,3 +234,26 @@ export const convertAnnualRateToSum = ({ date, value }) => {
     return value * 365;
   }
 };
+
+/**
+ * Create initial configuration
+ * @param {object} defaultParams - default params for tool
+ * @param {object} urlParams - params generated from url query string by sapper
+ * @return {object}
+ */
+// Helper function to create an object with initial settings for a Cal-Adapt tool
+export const getInitialConfig = (defaultParams = {}, urlParams = {}) => {
+  if (Object.keys(urlParams).length === 0) {
+    return defaultParams;
+  }
+
+  const { models, lat, lng, ...rest } = urlParams;
+
+  return {
+    ...defaultParams,
+    ...rest,
+    models: models.split(","),
```

Comment:
what if `models`, `lat`, or `lng` aren't in the `urlParams`? They will still override the values from the `defaultParams` but set them to `undefined` or `NaN` (I think).

Reply:
@clhenrick yes that's a tricky one. I added a check to see if the prop exists before adding it in https://github.com/berkeley-gif/caladapt-website-2021/pull/179/commits/1f423dce93618c7cff447a826c99087467555364